src/python/pants/backend/go/target_type_rules.py

```diff
@@ -210,6 +228,58 @@ async def inject_go_external_package_dependencies(
     return InjectedDependencies(inferred_dependencies)
 
 
+# -----------------------------------------------------------------------------------------------
+# Generate `_go_external_package` targets
+# -----------------------------------------------------------------------------------------------
+
+
+class GenerateGoExternalPackageTargetsRequest(GenerateTargetsRequest):
+    generate_from = GoModule
+
+
+@rule(desc="Generate targets for each external package in `go.mod`", level=LogLevel.DEBUG)
+async def generate_go_external_package_targets(
+    request: GenerateGoExternalPackageTargetsRequest,
+) -> GeneratedTargets:
+    generator_addr = request.generator.address
+    resolved_module = await Get(ResolvedGoModule, ResolveGoModuleRequest(generator_addr))
+    all_resolved_packages = await MultiGet(
+        Get(
+            ResolveExternalGoModuleToPackagesResult,
+            ResolveExternalGoModuleToPackagesRequest(
+                path=module_descriptor.path,
+                version=module_descriptor.version,
+                go_sum_digest=resolved_module.digest,
+            ),
+        )
+        for module_descriptor in resolved_module.modules
+    )
+
+    def create_tgt(pkg: ResolvedGoPackage) -> GoExternalPackageTarget:
+        return GoExternalPackageTarget(
+            {
+                GoExternalModulePathField.alias: pkg.module_path,
+                GoExternalModuleVersionField.alias: pkg.module_version,
+                GoExternalPackageImportPathField.alias: pkg.import_path,
+            },
+            # E.g. `src/go:mod#github.com/google/uuid`.
+            Address(
+                generator_addr.spec_path,
+                target_name=generator_addr.target_name,
+                generated_name=pkg.import_path,
```

Comment:
Based on https://golang.org/ref/spec#Import_declarations, it seems like the import path will never contain a `#`?